src/rust/include/rust/orchard/wallet.h

```diff
@@ -263,6 +263,59 @@ void orchard_wallet_get_filtered_notes(
         push_note_callback_t push_cb
         );
 
+/**
+ * A C struct used to transfer Orchard action spend information across the FFI boundary.
+ * This must have the same in-memory representation as the `FFIActionSpend` type in
+ * orchard_ffi/wallet.rs.
+ */
+struct RawOrchardActionSpend {
+    uint32_t spendActionIdx;
+    unsigned char outpointTxId[32];
+    uint32_t outpointActionIdx;
+    OrchardRawAddressPtr* receivedAt;
+    CAmount noteValue;
+};
+
+/**
+ * A C struct used to transfer Orchard action output information across the FFI boundary.
+ * This must have the same in-memory representation as the `FFIActionOutput` type in
+ * orchard_ffi/wallet.rs.
+ */
+struct RawOrchardActionOutput {
+    uint32_t outputActionIdx;
+    OrchardRawAddressPtr* addr;
+    CAmount noteValue;
+    unsigned char memo[512];
+    bool isOutgoing;
+};
+
+typedef void (*push_spend_t)(void* callbackReceiver, const RawOrchardActionSpend data);
+
+typedef void (*push_output_t)(void* callbackReceiver, const RawOrchardActionOutput data);
+
+/**
+ * Trial-decrypts the specfied Orchard bundle, and uses the provided callbacks to pass
+ * `RawOrchardActionSpend` and `RawOrchardActionOutput` values (corresponding to the
+ * actions of that bundle) to the provided result receiver.
+ *
+ * Note that the callbacks can perform any necessary conversion from a
+ * `RawOrchardActionSpend` or `RawOrchardActionOutput` value in addition to modifying the
+ * provided result receiver.
+ *
+ * `raw_ovks` must be a pointer to an array of `unsigned char[32]`.
+ *
+ * The `addr` pointer on each `RawOrchardActionOutput` value must be freed using
+ * `orchard_address_free`.
```

Comment:
What about the `receivedAt` pointer for each `RawOrchardActionSpend`?

Reply:
Fixed in #5715